src/wp-includes/rest-api.php

```diff
@@ -3389,3 +3389,39 @@ static function ( $status, $error_data ) {
 
 	return new WP_REST_Response( $data, $status );
 }
+
+/**
+ * Checks whether a REST API endpoint request is currently being handled.
+ *
+ * This may be a standalone REST API request, or an internal request dispatched from within a regular page load.
+ *
+ * @since 6.5.0
+ *
+ * @global WP_REST_Server $wp_rest_server REST server instance.
+ *
+ * @return bool True if a REST endpoint request is currently being handled, false otherwise.
+ */
+function wp_is_rest_endpoint(): bool {
```

Comment:
We could add a `$endpoint` parameter to this function to allow developers to check which endpoint if currently dispatched, like what is possible with [`is_post_type_archive`](is_post_type_archive).

```
if ( wp_is_rest_endpoint( 'wp/v2/post' ) ) {...}
```

Reply:
This could be done, but it's not in scope of this ticket/PR. Checking for a specific endpoint is a separate use-case, which I think should be explored in a separate ticket. It could be implemented later.